Homework7-start/Homework7/User/UserDetail.swift

```diff
@@ -0,0 +1,197 @@
+                      .font(.headline)
+                    Text(user.location.coordinates.longitude)
+                  }
+                }
+              }
+              Section (header: Text("Timezone")) {
+                HStack {
+                  Text("Offset: ")
+                    .font(.headline)
+                  Text(user.location.timezone.offset)
+                }
+                HStack {
+                  Text("Description: ")
+                    .font(.headline)
+                  Text(user.location.timezone.description)
+                }
+              }
+              Section(header: Text("Contact")) {
+                VStack(alignment: .leading) {
+                  HStack {
+                    Text("Phone:")
+                      .font(.headline)
+                    Text(user.phone)
+                  }
+                  HStack {
+                    Text("Cell: ")
+                      .font(.headline)
+                    Text(user.cell)
+                  }
+                  HStack {
+                    Text("Email: ")
+                      .font(.headline)
+                    Text(user.email)
+                  }
+                }
+              }
+              Section (header: Text("Login Information")) {
+                HStack(alignment: .top) {
+                  Text("UUID: ")
+                    .font(.headline)
+                  VStack {
+                    Text(user.login.uuid)
+                  }
+                }
+                HStack {
+                  Text("Username: ")
+                    .font(.headline)
+                  Text(user.login.username)
+                }
+                HStack {
+                  Text("Password: ")
+                    .font(.headline)
+                  SecurePassword(password: user.login.password)
+                }
+                HStack {
+                  Text("salt: ")
+                    .font(.headline)
+                  Text(user.login.salt)
+                }
+                HStack {
+                  Text("md5: ")
+                    .font(.headline)
+                  Text(user.login.md5)
+                }
+                HStack {
+                  Text("sha1: ")
+                  Text(user.login.sha1)
+                }
+                HStack {
+                  Text("sha256: ")
+                  Text(user.login.sha256)
+                }
+              }
+            }
+          }
+          .navigationTitle("User Info")
+        }
+      }
+    
+  
+  func formattedDate(from dateString: String) -> String {
+    let isoFormatter = ISO8601DateFormatter()
+    isoFormatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
+    
+    
+    if let date = isoFormatter.date(from: dateString) {
+      let dateFormatter = DateFormatter()
+      dateFormatter.dateStyle = .medium
+      dateFormatter.timeStyle = .none
+      return dateFormatter.string(from: date)
+    } else {
+      return dateString
+    }
+  }
```

Comment:
Not bad. But since this is only being used once you could make all this work on a computed property instead.